enterprise_subsidy/apps/api/v1/views.py

```diff
@@ -1 +1,216 @@
+        the permission required to be based conditionally on the type of action.
+        """
+        permission_for_action = {
+            "list": PERMISSION_CAN_READ_TRANSACTIONS,
+            "retrieve": PERMISSION_CAN_READ_TRANSACTIONS,
+            "create": PERMISSION_CAN_CREATE_TRANSACTIONS,
+            "reverse": PERMISSION_CAN_CREATE_TRANSACTIONS,
+            "update": PERMISSION_NOT_GRANTED,
+            "partial_update": PERMISSION_NOT_GRANTED,
+            "destroy": PERMISSION_NOT_GRANTED,
+        }
+        permission_required = permission_for_action.get(self.request_action, PERMISSION_NOT_GRANTED)
+        return permission_required
+
+    @property
+    def requested_enterprise_customer_uuid(self):
+        """
+        Look in the query parameters for an enterprise customer UUID.
+        """
+        return utils.get_enterprise_uuid_from_request_query_params(self.request)
+
+    @property
+    def requested_transaction_uuid(self):
+        """
+        Fetch the transaction UUID from the URL location.
+
+        For detail endpoints, the PK can simply be found in ``self.kwargs``.
+        """
+        return self.kwargs.get('uuid')
+
+    @property
+    def base_queryset(self):
+        """
+        Required by the ``PermissionRequiredForListingMixin``.
+        For non-list actions, this is what's returned by ``get_queryset()``.
+        For list actions, some non-strict subset of this is what's returned by ``get_queryset()``.
+        """
+        kwargs = {}
+        if self.requested_enterprise_customer_uuid:
+            kwargs.update({'ledger__subsidy__enterprise_customer_uuid': self.requested_enterprise_customer_uuid})
+        if self.requested_transaction_uuid:
+            kwargs.update({'uuid': self.requested_transaction_uuid})
+
+        return Transaction.objects.filter(**kwargs).order_by('uuid')
+
+    def update(self, *args, **kwargs):
+        """
+        We should never get to this point because the permission for this action is not granted, but just in case we'll
+        force a 403 response.
+        """
+        response = {'message': 'Update action is not offered.'}
+        return Response(response, status=status.HTTP_403_FORBIDDEN)
+
+    def partial_update(self, *args, **kwargs):
+        """
+        We should never get to this point because the permission for this action is not granted, but just in case we'll
+        force a 403 response.
+        """
+        response = {'message': 'Update action is not offered.'}
+        return Response(response, status=status.HTTP_403_FORBIDDEN)
+
+    def destroy(self, *args, **kwargs):
+        """
+        We should never get to this point because the permission for this action is not granted, but just in case we'll
+        force a 403 response.
+        """
+        response = {'message': 'Destroy action is not offered.'}
+        return Response(response, status=status.HTTP_403_FORBIDDEN)
```

Comment:
You can probably use this example mixin inheritance to achieve the same thing: https://www.django-rest-framework.org/api-guide/viewsets/#custom-viewset-base-classes